pyomo/core/base/symbolic.py

```diff
@@ -8,123 +8,15 @@
-        sympy.exp: lambda x: core.exp(x),
-        sympy.log: lambda x: core.log(x),
-        sympy.sin: lambda x: core.sin(x),
-        sympy.asin: lambda x: core.asin(x),
-        sympy.sinh: lambda x: core.sinh(x),
-        sympy.asinh: lambda x: core.asinh(x),
-        sympy.cos: lambda x: core.cos(x),
-        sympy.acos: lambda x: core.acos(x),
-        sympy.cosh: lambda x: core.cosh(x),
-        sympy.acosh: lambda x: core.acosh(x),
-        sympy.tan: lambda x: core.tan(x),
-        sympy.atan: lambda x: core.atan(x),
-        sympy.tanh: lambda x: core.tanh(x),
-        sympy.atanh: lambda x: core.atanh(x),
-        sympy.ceiling: lambda x: core.ceil(x),
-        sympy.floor: lambda x: core.floor(x),
-        sympy.sqrt: lambda x: core.sqrt(x),
-        sympy.Derivative: _nondifferentiable,
-        sympy.Tuple: lambda *x: x,
-    }
-
-    _pyomo_operator_map = {
-        EXPR.SumExpression: sympy.Add,
-        EXPR.ProductExpression: sympy.Mul,
-        EXPR.NPV_ProductExpression: sympy.Mul,
-        EXPR.MonomialTermExpression: sympy.Mul,
-    }
-
-    _functionMap = {
-        'exp': sympy.exp,
-        'log': sympy.log,
-        'log10': lambda x: sympy.log(x)/sympy.log(10),
-        'sin': sympy.sin,
-        'asin': sympy.asin,
-        'sinh': sympy.sinh,
-        'asinh': sympy.asinh,
-        'cos': sympy.cos,
-        'acos': sympy.acos,
-        'cosh': sympy.cosh,
-        'acosh': sympy.acosh,
-        'tan': sympy.tan,
-        'atan': sympy.atan,
-        'tanh': sympy.tanh,
-        'atanh': sympy.atanh,
-        'ceil': sympy.ceiling,
-        'floor': sympy.floor,
-        'sqrt': sympy.sqrt,
-    }
-except ImportError:
-    _sympy_available = False
-
-# A "public" attribute indicating that differentiate() can be called
-# ... this provides a bit of future-proofing for alternative approaches
-# to symbolic differentiation.
-differentiate_available = _sympy_available
-
-class NondifferentiableError(ValueError):
-    """A Pyomo-specific ValueError raised for non-differentiable expressions"""
-    pass
-
-class PyomoSympyBimap(object):
-    def __init__(self):
-        self.pyomo2sympy = ComponentMap()
-        self.sympy2pyomo = {}
-        self.i = 0
-
-    def getPyomoSymbol(self, sympy_object, default=None):
-        return self.sympy2pyomo.get(sympy_object, default)
-
-    def getSympySymbol(self, pyomo_object):
-        if pyomo_object in self.pyomo2sympy:
-            return self.pyomo2sympy[pyomo_object]
-        sympy_obj = sympy.Symbol("x%d" % self.i)
-        self.i += 1
-        self.pyomo2sympy[pyomo_object] = sympy_obj
-        self.sympy2pyomo[sympy_obj] = pyomo_object
-        return sympy_obj
-
-    def sympyVars(self):
-        return iterkeys(self.sympy2pyomo)
 
+@deprecated(msg=('The differentiate function in pyomo.core.base.symbolic has been deprecated. Please use the ' +
+                 'differentiate function in pyomo.core.expr.differentiate.differentiate'),
+            version='5.6.6',
```

Comment:
The version should be "TBD".  we will set it to the next release as part of the Pyomo release process.